workflows/cloudify_system_workflows/snapshots/agents.py

```diff
@@ -51,16 +51,19 @@ def restore(self, tempdir, version):
     def dump(self, tempdir, manager_version):
         self._manager_version = manager_version
         result = {}
-        for tenant_name in get_tenants_list():
-            result[tenant_name] = {}
-            tenant_client = get_rest_client(tenant_name)
-            tenant_deployments = tenant_client.deployments.list(
-                _include=['id'],
-                _get_all_results=True
-            )
-            for deployment in tenant_deployments:
-                result[tenant_name][deployment.id] = \
-                    self._get_deployment_result(tenant_client, deployment.id)
+        client = get_rest_client()
+        agents = client.agents.list(all_tenants=True)
+        node_instances = list(client.node_instances.list(
+            id=[item['id'] for item in agents.items],
+            _all_tenants=True))
+        for inst in node_instances:
+            result.setdefault(inst['tenant_name'], {})
```

Comment:
Maybe all the content of the `for` should be in another function `add_instance_to_result`